cpp/include/raft_runtime/neighbors/ivf_pq.hpp

```diff
@@ -40,33 +49,53 @@ RAFT_INST_SEARCH(uint8_t, uint64_t);
 // We define overloads for build and extend with void return type. This is used in the Cython
 // wrappers, where exception handling is not compatible with return type that has nontrivial
 // constructor.
-#define RAFT_INST_BUILD_EXTEND(T, IdxT)                               \
-  auto build(raft::device_resources const& handle,                    \
-             const raft::neighbors::ivf_pq::index_params& params,     \
-             const T* dataset,                                        \
-             IdxT n_rows,                                             \
-             uint32_t dim)                                            \
-    ->raft::neighbors::ivf_pq::index<IdxT>;                           \
-                                                                      \
-  auto extend(raft::device_resources const& handle,                   \
-              const raft::neighbors::ivf_pq::index<IdxT>& orig_index, \
-              const T* new_vectors,                                   \
-              const IdxT* new_indices,                                \
-              IdxT n_rows)                                            \
-    ->raft::neighbors::ivf_pq::index<IdxT>;                           \
-                                                                      \
-  void build(raft::device_resources const& handle,                    \
-             const raft::neighbors::ivf_pq::index_params& params,     \
-             const T* dataset,                                        \
-             IdxT n_rows,                                             \
-             uint32_t dim,                                            \
-             raft::neighbors::ivf_pq::index<IdxT>* idx);              \
-                                                                      \
-  void extend(raft::device_resources const& handle,                   \
-              raft::neighbors::ivf_pq::index<IdxT>* idx,              \
-              const T* new_vectors,                                   \
-              const IdxT* new_indices,                                \
-              IdxT n_rows);
+#define RAFT_INST_BUILD_EXTEND(T, IdxT)                                                 \
+  auto build(raft::device_resources const& handle,                                      \
```

Comment:
it's okay to keep the pointer-based specializations but we should remove the pointer-based functions from `raft::runtime`.